src/__tests__/cache.test.js

```diff
@@ -89,48 +132,114 @@ describe('createCachingMethods', () => {
     expect(collection.find.mock.calls.length).toBe(1)
   })
 
-  // TODO why doesn't this pass?
-  // it.only(`doesn't cache without ttl`, async () => {
-  //   await api.findOneById(docs.id1._id)
-  //   await api.findOneById(docs.id1._id)
+  it('finds by field', async () => {
+    const foundDocs = await api.findByFields({ foo: 'bar' })
+
+    expect(foundDocs[0]).toBe(docs.one)
+    expect(foundDocs[1]).toBe(docs.two)
+    expect(foundDocs.length).toBe(2)
+
+    expect(collection.find.mock.calls.length).toBe(1)
+  })
+
+  it('finds by array field', async () => {
+    const foundDocs = await api.findByFields({ tags: 'bar' })
+
+    expect(foundDocs[0]).toBe(docs.one)
+    expect(foundDocs[1]).toBe(stringDoc)
+    expect(foundDocs.length).toBe(2)
+
+    expect(collection.find.mock.calls.length).toBe(1)
+  })
+
+  it('finds by mutiple fields', async () => {
+    const foundDocs = await api.findByFields({
+      tags: ['foo', 'bar'],
+      foo: 'bar'
+    })
+
+    expect(foundDocs[0]).toBe(docs.one)
+    expect(foundDocs.length).toBe(1)
+
+    expect(collection.find.mock.calls.length).toBe(1)
+  })
+
+  it(`doesn't mix filters of pending calls for different fields`, async () => {
+    const pendingDocs1 = api.findByFields({ foo: 'bar' })
+    const pendingDocs2 = api.findByFields({ tags: 'baz' })
+    const [foundDocs1, foundDocs2] = await Promise.all([
+      pendingDocs1,
+      pendingDocs2
+    ])
+
+    expect(foundDocs1[0]).toBe(docs.one)
+    expect(foundDocs1[1]).toBe(docs.two)
+    expect(foundDocs1.length).toBe(2)
+    expect(foundDocs2[0]).toBe(stringDoc)
+    expect(foundDocs2.length).toBe(1)
+
+    expect(collection.find.mock.calls.length).toBe(1)
+  })
+
+  it(`dataloader caches each value individually when finding by a single field`, async () => {
+    await api.findByFields({ tags: ['foo', 'baz'] }, { ttl: 1 })
+
+    const fooBazCacheValue = await cache.get(
```

Comment:
`cache` is separate from dataloader. i suggest removing "dataloader" from the test title and creating other tests for dataloader (which should do memoization caching: https://github.com/graphql/dataloader#caching)

Reply:
This is testing the memoization caching. The desired behavior is that if `findByFields` is used with a single field and multiple values, each individual value should get cached by the dataloaders memoization cache (tested by calling with 2 values, 'foo' and 'baz', on line 185, then calling with a single value, 'foo', on line 195, then expecting `find` to have been called only once on line 196). However, the `InMemoryLRUCache` cache should not be affected by this (tested by ensuring the cache has a result for both values on line 192, but not for a single value, on line 193)